test/hotspot/jtreg/compiler/c2/TestMergeLoads.java

```diff
@@ -0,0 +1,706 @@
+                verifyS(name, i, (short[])g, (short[])r);
+            } else if (c == int.class) {
+                verifyI(name, i, (int[])g, (int[])r);
+            } else if (c == long.class) {
+                verifyL(name, i, (long[])g, (long[])r);
+            } else {
+                throw new RuntimeException("verify " + name + ": array type not supported for verify:" +
+                                       " gold[" + i + "].getClass() = " + g.getClass().getSimpleName() +
+                                       " result[" + i + "].getClass() = " + r.getClass().getSimpleName());
+            }
+        }
+    }
+
+    static void verifyB(String name, int i, byte[] g, byte[] r) {
+        for (int j = 0; j < g.length; j++) {
+            if (g[j] != r[j]) {
+                throw new RuntimeException("verify " + name + ": arrays must have same content:" +
+                                           " gold[" + i + "][" + j + "] = " + g[j] +
+                                           " = " + String.format("%02X", g[j] & 0xFF) +
+                                           " result[" + i + "][" + j + "] = " + r[j] +
+                                           " = " + String.format("%02X", r[j] & 0xFF));
+            }
+        }
+    }
+
+    static void verifyS(String name, int i, short[] g, short[] r) {
+        for (int j = 0; j < g.length; j++) {
+            if (g[j] != r[j]) {
+                throw new RuntimeException("verify " + name + ": arrays must have same content:" +
+                                           " gold[" + i + "][" + j + "] = " + g[j] +
+                                           " result[" + i + "][" + j + "] = " + r[j]);
+            }
+        }
+    }
+
+    static void verifyI(String name, int i, int[] g, int[] r) {
+        for (int j = 0; j < g.length; j++) {
+            if (g[j] != r[j]) {
+                throw new RuntimeException("verify " + name + ": arrays must have same content:" +
+                                           " gold[" + i + "][" + j + "] = " + g[j] +
+                                           " result[" + i + "][" + j + "] = " + r[j]);
+            }
+        }
+    }
+
+    static void verifyL(String name, int i, long[] g, long[] r) {
+        for (int j = 0; j < g.length; j++) {
+            if (g[j] != r[j]) {
+                throw new RuntimeException("verify " + name + ": arrays must have same content:" +
+                                           " gold[" + i + "][" + j + "] = " + g[j] +
+                                           " result[" + i + "][" + j + "] = " + r[j]);
+            }
+        }
+    }
+
+    /**
+     * Group 1: get int in little endian mode
+     */
+    @DontCompile
+    static Object[] test1R(byte[] a) {
+      int i1 = (a[0] & 0xff)         |
+              ((a[1] & 0xff) << 8 )  |
+              ((a[2] & 0xff) << 16)  |
+              ((a[3] & 0xff) << 24);
+      int[] ret = {i1};
+      return new Object[]{ret};
+    }
```

Comment:
Ah, I see you are using `|`. Can we also use `+`?

FYI: if you use `Verify.java`, then you could directly do `return i2;`. It would get returned as a boxed `Integer`, and should be compared that way, without allocating an `int[]` and `Object[]`.

Reply:
Now I only check ‘|’, so they can not be merged if using '+'. But I think it can be extended.